server/hotrod/src/main/scala/org/infinispan/server/hotrod/iteration/IterationFilters.scala

```diff
@@ -22,10 +22,10 @@ import scala.util._
 
 class IterationFilter[K, V, Any](val compat: Boolean,
                                  val providedFilter: Option[KeyValueFilterConverter[K, V, Any]],
-                                 val segments: Bytes,
+                                 val segments: Option[Bytes],
```

Comment:
Since ```providedFilter``` param was already resolved by the caller into a ```KeyValueFilterConverter``` why not make ```segments``` a ```Option[java.util.BitSet]``` too?